source/MaterialXRuntime/Private/Commands/PvtRemoveMetadataCmd.cpp

```diff
@@ -0,0 +1,111 @@
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRuntime/Private/Commands/PvtRemoveMetadataCmd.h>
+#include <MaterialXRuntime/Private/PvtCommand.h>
+
+#include <MaterialXRuntime/RtPrim.h>
+#include <MaterialXRuntime/RtTypeDef.h>
+#include <MaterialXRuntime/RtValue.h>
+
+namespace MaterialX
+{
+
+PvtRemoveMetadataCmd::PvtRemoveMetadataCmd(RtObject& obj, const RtToken& name)
+    : PvtCommand()
+    , _obj(obj)
+    , _name(name)
+    , _oldValue()
+{
+}
+
+PvtCommandPtr PvtRemoveMetadataCmd::create(RtObject& obj, const RtToken& name)
+{
+    return std::make_shared<PvtRemoveMetadataCmd>(obj, name);
+}
+
+void PvtRemoveMetadataCmd::execute(RtCommandResult& result)
+{
+    if (_obj.isValid())
+    {
+        try
+        {
+            // Send message that the metadata is being removed
+            msg().sendRemoveMetadataMessage(_obj, _name);
+
+            // Save old value for undo/redo
+            _oldValue = RtValue::clone(RtType::STRING, _obj.getMetadata(_name)->getValue(), _obj.getParent());
+
+            // Remove the metadata value
+            _obj.removeMetadata(_name);
+            result = RtCommandResult(true);
+        }
+        catch (const ExceptionRuntimeError& e)
+        {
+            result = RtCommandResult(false, string(e.what()));
+        }
+    }
+    else
+    {
+        result = RtCommandResult(false, string("Node to remove metadata from is no longer valid"));
+    }
+}
+
+void PvtRemoveMetadataCmd::undo(RtCommandResult& result)
+{
+    if (_obj.isValid())
+    {
+        try
+        {
+            // Send message that the metadata is being created
+            msg().sendCreateMetadataMessage(_obj, _name, _oldValue);
+
+            RtTypedValue* md = _obj.addMetadata(_name, RtType::STRING);
+            if (md)
+            {
+                md->setValue(_oldValue);
+                result = RtCommandResult(true);
+            }
+            else
+            {
+                result = RtCommandResult(false, "Unable to create metadata");
+            }
+        }
+        catch (const ExceptionRuntimeError& e)
+        {
+            result = RtCommandResult(false, string(e.what()));
+        }
+    }
+    else
+    {
+        result = RtCommandResult(false, string("Node to add metadata to is no longer valid"));
+    }
+}
+
+void PvtRemoveMetadataCmd::redo(RtCommandResult& result)
+{
+   if (_obj.isValid())
+    {
+        try
+        {
+            // Send message that the metadata is being removed
+            msg().sendRemoveMetadataMessage(_obj, _name);
+
+            // Remove the metadata value
+            _obj.removeMetadata(_name);
+            result = RtCommandResult(true);
+        }
+        catch (const ExceptionRuntimeError& e)
+        {
+            result = RtCommandResult(false, string(e.what()));
+        }
+    }
+    else
+    {
+        result = RtCommandResult(false, string("Node to remove metadata from is no longer valid"));
```

Comment:
Nitpick: Maybe say Object instead of Node, since this applied to any object and not just nodes.